kit/inputs/src/UIMaterialTextView/UIMaterialTextView.tsx

```diff
@@ -62,19 +59,36 @@ function useExtendedProps(
         isFocused,
     );
 
+    const { selectionEnd, onSelectionChange, skipNextOnSelectionChange } = useOnSelectionChange();
+
+    const applyMask = useApplyMask(mask, selectionEnd, skipNextOnSelectionChange);
+
     const { imperativeChangeText, moveCarret } = useImperativeChange(
         ref,
-        multiline,
-        onChangeTextWithInputHasValue,
+        onChangeTextProp,
+        checkInputHasValue,
+        applyMask,
     );
 
-    const { onChangeText, onSelectionChange } = useApplyMask(
-        imperativeChangeText,
-        moveCarret,
-        mask,
+    const onChangeText = React.useCallback(
+        (text: string) =>
+            imperativeChangeText(text, {
+                shouldSetNativeProps: false,
+            }),
+        [imperativeChangeText],
     );
 
-    const clear = useClear(resetInputHeight, onChangeText, ref);
+    React.useEffect(() => {
+        if (defaultValue) {
+            imperativeChangeText(defaultValue);
```

Comment:
But how do you guarantee a consistency between mount and `useEffect`?

Reply:
Because between those two states input will have an incorrect value

Reply:
Why I need guarantee this consistency? Update takes tiny time. I don't think we need to make it less. But this way guarantee ref to be assigned, that needed to proper work of this component.

Reply:
I can use useLayoutEffect to make this calculations before first render without blink.
Putting defaultValue to nested TextInput is wrong because first of all defaultValue should be validated by mask.

Reply:
`useLayoutEffect` doesn't make calculations before first render, in fact it actually can fire after `useEffect` :)

Reply:
Done